client/src/components/table/Table.tsx

```diff
@@ -0,0 +1,95 @@
+// Importing dependencies.
+import { useState, useEffect } from "react";
+
+// Importing api.
+import { fetchCustomers } from "../../api/customers";
+
+// Importing scripts.
+import formatDate from "../../utils/formatDate";
+
+// Importing interfaces.
+import type Customer from "../../interfaces/customer";
+
+export default function Table() {
+  const [page, setPage] = useState(1);
+  const [limit, setLimit] = useState(10);
+  const [customers, setCustomers] = useState([]);
+  const [allCustomers, setAllCustomers] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      const customersData = await fetchCustomers();
+      setAllCustomers(customersData);
```

Comment:
```suggestion
      setAllCustomers(customersData?.payload || []);
```

L'endpoint dei clienti restituisce il risultato in formato oggetto sotto chiave `payload`, pertanto senza questa aggiunta non sarebbe possibile visualizzare i clienti e si avrebbe errore successivamente nello slice.

Reply:
Un altro approccio poteva essere quello di cambiare la funzione in `/src/api/customers.ts` in modo che tenesse conto di payload.